jsonrpc/txpool_endpoint.go

```diff
@@ -0,0 +1,153 @@
+package jsonrpc
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/0xPolygon/minimal/types"
+)
+
+// Txpool is the txpool jsonrpc endpoint
+type Txpool struct {
+  d *Dispatcher
+}
+
+type ContentResponse struct {
+  Pending   map[types.Address]map[uint64]*gethTransaction   `json:"pending"`
+  Queued    map[types.Address]map[uint64]*gethTransaction   `json:"queued"`
+}
+
+type InspectResponse struct {
+  Pending   map[string]map[string]string  `json:"pending"`
+  Queued    map[string]map[string]string  `json:"queued"`
+}
+
+type StatusResponse struct {
+  Pending   uint64  `json:"pending"`
+  Queued    uint64  `json:"queued"`
+}
+
+type gethTransaction struct {
+  Nonce       argUint64      `json:"nonce"`
+  GasPrice    argBig         `json:"gasPrice"`
+  Gas         argUint64      `json:"gas"`
+  To          *types.Address `json:"to"`
+  Value       argBig         `json:"value"`
+  Input       argBytes       `json:"input"`
+  V           argByte        `json:"v"`
+  R           argBytes       `json:"r"`
+  S           argBytes       `json:"s"`
+  Hash        types.Hash     `json:"hash"`
+  From        types.Address  `json:"from"`
+  BlockHash   types.Hash     `json:"blockHash"`
+  BlockNumber interface{}    `json:"blockNumber"`
+  TxIndex     interface{}    `json:"transactionIndex"`
+}
+
+func toGethTransaction(t *types.Transaction) *gethTransaction {
+  if t.R == nil {
```

Comment:
Here if R or S is `nil`, argBytes will format the response json as `0x` however this isn't an acceptable value. Therefore, I initialized a 0 byte array if either of these fields are `nil`.

Reply:
Thank you for the explanation. Could you add it to the code as well, since these comments will not be visible in the future when someone is going over it?